runtime-sdk/src/dispatcher.rs

```diff
@@ -102,27 +103,53 @@ impl<R: Runtime> Dispatcher<R> {
             .map_err(|_| modules::core::Error::MalformedTransaction)
     }
 
-    fn dispatch_tx(
+    fn dispatch_call(
         &self,
-        ctx: &mut DispatchContext<'_>,
-        tx: types::transaction::Transaction,
-    ) -> Result<DispatchResult, Error> {
-        // Run pre-processing hooks.
-        if let Err(err) = R::Modules::authenticate_tx(ctx, &tx) {
-            return Ok(err.to_call_result().into());
+        mut ctx: &mut TxContext<'_, '_>,
+        call: types::transaction::Call,
+    ) -> types::transaction::CallResult {
+        // Use up gas for authenticating the transaction.
+        if let Err(e) = modules::core::Module::use_gas_for_auth(&mut ctx) {
+            return e.to_call_result();
```

Comment:
Hm, any particular reason why you didn't just add this same call to the previous implementation of an "estimate gas" query instead of turning all the things around?

Reply:
lower risk that they'd accidentally diverge if we make a mistake maintaining these two somewhat-far-away copies of the same steps